doc/application/application.rst

```diff
@@ -161,53 +510,102 @@ build scripts like this:
-source file src/main.c, behaviour that we surely do not want. The
-'PUBLIC' keyword could however be useful when modifying the include
+In the above code, an existing library target named ``app`` is
+configured to include the source file :file:`src/main.c`. The ``PRIVATE``
+keyword indicates that we are modifying the internals of how the library is
+being built. Using the keyword ``PUBLIC`` would modify how other
+libraries that link with app are built. In this case, using ``PUBLIC``
+would cause libraries that link with ``app`` to also include the
+source file :file:`src/main.c`, behaviour that we surely do not want. The
+``PUBLIC`` keyword could however be useful when modifying the include
 paths of a target library.
 
 Application CMakeLists.txt
-**************************
+==========================
+
+Every application must have a :file:`CMakeLists.txt` file. This
+file is the entry point, or top level, of the build
+system that builds the final :file:`zephyr.elf` image. :file:`zephyr.elf`
+contains both the application and the kernel libraries.
+
+This section describes some of what you can do in your :file:`CMakeLists.txt`.
+Make sure to follow these steps in order.
+
+#. If you only want to build for one board, add the name of the board
+   configuration for your application on a new line. For example:
+
+   .. code-block:: cmake
+
+      set(BOARD qemu_x86)
+
+   Refer to :ref:`boards` for more information on available boards.
+
+   The Zephyr build system determines a value for :makevar:`BOARD` by checking
+   the following, in order:
+
+   - Any previously used value as determined by the CMake cache takes highest
+     precedence. This ensures you don't try to run a build with a different
+     :makevar:`BOARD` value than you set during the build configuration step.
+
+   - Any value given on the CMake command line using ``-DBOARD=YOUR_BOARD``
+     will be checked for and used next.
+
+   - If an environment variable ``BOARD`` is set, its value will then be used.
+
+   - Finally, if you set ``BOARD`` in your application :file:`CMakeLists.txt`
+     as described in this step, this value will be used.
+
+#. If your application uses a configuration file or files other than
+   the usual :file:`prj.conf` (or :file:`prj_YOUR_BOARD.conf`, where
+   ``YOUR_BOARD`` is a board name), you can now add lines setting the
+   :makevar:`CONF_FILE` variable to these files appropriately.
+
+   More details are available below in :ref:`application_configuration`.
+
+#. If your application has its own kernel configuration options, you can add a
+   line setting the location of the Kconfig file which defines them.
 
-Every application must have a CMakeLists.txt file. The application
-CMakeLists.txt file is the entry point, or toplevel of the build
-system that builds the :file:`zephyr.elf` image. :file:`zephyr.elf`
-contains both the application and the kernel libraries.
+   **This is a somewhat advanced use case; you probably don't need this.**
```

Comment:
Change to:

     An (unlikely) advanced use case would be if your application has its own
     unique configuration **options** that are set differently depending on the
     build configuration.

Then continue with, "If you just want to ..."

Reply:
OK, will fix.